src/encoder.rs

```diff
@@ -2033,7 +2100,7 @@ pub fn encode_block_with_modes<T: Pixel>(
   fi: &FrameInvariants<T>, ts: &mut TileStateMut<'_, T>,
   cw: &mut ContextWriter, w_pre_cdef: &mut dyn Writer,
   w_post_cdef: &mut dyn Writer, bsize: BlockSize, tile_bo: TileBlockOffset,
-  mode_decision: &RDOPartitionOutput, rdo_type: RDOType,
+  mode_decision: &RDOPartitionOutput, rdo_type: RDOType, record_stats: bool,
```

Comment:
it should go in `Config`, not be a parameter.

Reply:
I don't think it could, because it depends on where we're calling `encode_block` from. If we're calling from RDO analysis, we don't want to record stats because that's a trial encode. We only want to record stats for the final block encode.

Reply:
You are right.